# SimPanel
Simulating a tumour gene panel for NGS

The main purpose is to provide a well-defined read set that can be used as a test for panel workflows. It contains no data from real patients, can be modified, etc. 

Software/data used:
 - HG19 FASTA
 - bedtools
 - wgsim
 - samtools
 - [bamsurgeon](https://github.com/adamewing/bamsurgeon) and all of its dependencies but Picard, namely:
   - bwa
   - velvet 
   - exonerate 
   - pysam

##Steps to build the normal dataset:

###Create reference fasta
The bed file `panel\_hg19.bed` contains gene regions to be cut from the HG19 FASTA file. These coordinates were actually lifted over from 
GRCh38, so should not be treated as a concise list of  tumour genes.

```
 bedtools getfasta -fi hg19.fa -bed panel\_hg19.bed.bed > tumor_panel_37.fasta
```

###Generate normal reads
Reads to be treated as normal can be generated by `wgsim`. The size of the reference is about 200M still, it is really a large panel having 1380 genes:

```
$ grep -v ">" tumor_panel_37.fasta | wc -c
191942266
```
We are making 70M read pairs each read having 151 bps length and including some germline variations (storing these in variations.txt):
```
wgsim -N 70000000 -1 151 -2 151 tumor_panel_37.fasta tumor_panel_37_R1.fastq tumor_panel_37_R2.fastq 2>&1 | tee variations.txt
```
While reference\_length\*coverage/(2\*readlength) = number\_of\_reads, therefore, coverage = 2\*readlength\*number\_of\_reads/reference\_length

2x151x70000000/191942266 = 110

coverage can be achieved. 
